### ISSUE TITLE
react-oidc compatibility with React 19

### ISSUE THREAD

**Issue by awocallaghan** [AxaFrance/oidc-client#1370]
### Issue and Steps to Reproduce

If you try to use the React 19 beta version with the `react-oidc` package there's an error:

```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'ReactCurrentDispatcher')
```

This is because `react-oidc` includes `react-jsx-runtime` in the library build.

I was able to get this working by changing the [vite.config.ts](https://github.com/AxaFrance/oidc-client/blob/main/packages/react-oidc/vite.config.ts#L20) `build.rollupOptions.external` to include `"react/jsx-runtime"`:

```ts
rollupOptions: {
      external: [
        ...dependencies,
        'react/jsx-runtime',
      ],
      ....
}
```

### Versions

7.22.4

#### Expected

React and related dependencies to be treated as external / peer dependencies

#### Actual

`react-jsx-runtime` included in build, causing errors when using React 19 beta

### Additional Details

React 19 beta upgrade guide: https://react.dev/blog/2024/04/25/react-19

**Comment by guillaume-chervet** [comment]
Thank you very much @awocallaghan for your issue and the way to resolved it !
I wil try to do it as quickly as I can.

**Comment by guillaume-chervet** [comment]
hi @awocallaghan , I published 7.22.6-alpha.1437 with the change you propose.

**Comment by awocallaghan** [comment]
> hi @awocallaghan , I published 7.22.6-alpha.1437 with the change you propose.

Amazing! Thanks for all your work maintaining this
